### DISCUSSION TITLE
Streamer support

### DISCUSSION THREAD

**Discussion by SporyTike** [RetroAchievements/RAWeb#768] [Ideas]
A while ago I thought about a support for RA streamers and the last few weeks I also heard it from streamers itself.
Currently there are a few RA users who stream their achievement earnings and development and as a result they do advertising for the site because some viewers see it and join RA. In my opinion it would be fair to support those streamers.
In the past there was a stream section for the RA Twitch channel on the mainpage so I thought about bring something like this back. A small section where people can see which users stream RA at the moment (maybe with game).

**Comment by Thoooreau** [comment]
Maybe a page like http://www.speedrunslive.com/, right? Would be cool, indeed.

**Comment by televandalist** [comment]
I brought up having featured streamers at a staff meeting a while back and everyone was on board. It's just a matter of someone volunteering to take charge of it and keep it organized. 

FYI: I'm pretty ignorant about streaming culture, but know many people at RA enjoy it.

**Comment by SporyTike** [comment]
I was talking with Wes about this earlier today and we think we should really get this done since many people advertise RetroAchievements, some which even have multiple hundrets of viewers. The only way to advertise your stream at the moment is via the Discord channel which isn't the best since only people who are on the server can see the list, your account needs to be on it,  the channel gets flooded due to embeds and you can't see who is still live and who stopped streaming.

A subpage would really do a great job. I made an image to show how it could look like:
![image](https://github.com/RetroAchievements/RAWeb/assets/45672834/5400239d-798a-4cfe-8245-44baa9c270ae)
After some discussion we came up with some things to consider while developing.

1. **Subpage showing active streams**
Basically everything related to the subpage and to make it work. Likely we could go with the same idea as the bot, checking for current streams with the tag 'RetroAchievements' using the Twitch API. A list of active streams can currently found on Twitch directly using this link: https://www.twitch.tv/directory/all/tags/RetroAchievements

2. **Connect Twitch account (optional)**
If there would be a way to connect a Twitch account with the RA account, it would be possible to grab data for the subpage like current game/set and progression on the set by grabbing the most recent RP. There could either be a direct Twitch integration or maybe even something as simple as a text field in which you enter the Twitch name.

3. **Add Highlight List (optional)**
There should be a highlight list that can be edited by site moderators. Every once in a while we have some special events that get livestreamed like RetroOlympics or maybe anniversary streams of longtime users. This would allow to place the highlighted streams above all others, so that they don't get buried by streams with more viewers or whatever kind of sorting there would be.

4. **Add Blacklist (optional)**
A general blacklist should exist so that we can exclude specific Twitch accounts and hide them on the subpage. This can be useful in case someone abuses the tag without playing RetroAchievements or if we simply don't want to support streamers for a specific reason like extreme bad language, hate speech or they tend to be a trigger happy.

5. **Other Platforms (optional)**
While Twitch is probably the main platform people stream RetroAchievements on, there might be some interested in streaming on other pages like YouTube and it would be nice if those could be shown too.

**Reply by hb-robo** [reply]
Wow, this is a really cool idea. I love the integrated user progress.

**Reply by Hexadigital** [reply]
> This can be useful in case someone abuses the tag without playing RetroAchievements

Is this referring to things like !ra in the title or the actual RetroAchievements tag? If the latter, most streamers don't change their tags on a stream-to-stream basis, so it might be better to handle it how the Discord bot currently handles it and look for !ra or RetroAchievements in the stream title.

**Reply by christianlegge** [reply]
Connecting the Twitch account is a great idea, showing the current progress under the stream is really fun. A couple points:

1. It needs to be a for-real Twitch integration - if it was just a text field you could get someone registering an account just to take someone else's Twitch name, you could have someone deliberately putting someone else's stream just to inject incorrect RP data, you could have typos... the integration solves these problems, that's what it's for.
2. There should be some kind of time-based check to make sure their RP has updated recently enough to associate it with the stream. Naively you might think that you could use the Twitch category to verify that they're playing, but many RA streamers won't bother setting the category to the specific game, rather using Games + Demos or Retro (especially for the obscurer ones).

**Reply by Hexadigital** [reply]
> Naively you might think that you could use the Twitch category to verify that they're playing, but many RA streamers won't bother setting the category to the specific game, rather using Games + Demos or Retro (especially for the obscurer ones).

To add to this, many of the less-notable games on RA with sets (especially homebrew and hacks) do not exist in Twitch's category list at all. Going by stream title, like is currently done for the RA Discord, would be my recommendation.

**Reply by LiftedSenses** [reply]
This is a great idea for the site. It will help more RA streamers be discovered.

**Reply by AmericanSlacker** [reply]
I agree, this is a cool idea.

**Reply by definesfineout** [reply]
I also agree this would be a great addition to the site. I'd love to see this implemented.

**Reply by GeeksOfRage** [reply]
I dropped a brief comment on discord about youtube streamers like myself not having a place for our RA streams. If this could be something that gets created, that would be great. I'm also in production of guides and tips videos for games I have either mastered or "hard" achievements I unlocked.
